lib/msal-common/src/authority/RegionDiscovery.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { INetworkModule } from "../network/INetworkModule";
+import { NetworkResponse } from "../network/NetworkManager";
+import { IMDSBadResponse } from "../response/IMDSBadResponse";
+import { Constants } from "../utils/Constants";
+
+export class RegionDiscovery {
+    // Network interface to make requests with.
+    protected networkInterface: INetworkModule;
+    // Options for the IMDS endpoint request
+    protected static IMDS_OPTIONS = {headers: {"Metadata": "true"}};
+
+    constructor(networkInterface: INetworkModule) {
+        this.networkInterface = networkInterface;
+    }
+
+    /**
+     * Detect the region from the application's environment.
+     * 
+     * @returns Promise<string | null>
+     */
+    public async detectRegion(environmentRegion: string | undefined): Promise<string | null> {
+        // Initialize auto detected region with the region from the envrionment 
+        let autodetectedRegionName = environmentRegion;
+
+        // Call the local IMDS endpoint for applications running in azure vms
+        if (!autodetectedRegionName) {
+            try {
+                const response = await this.getRegionFromIMDS(Constants.IMDS_VERSION);
+                if (response.status === 200) {
```

Comment:
can we have constants:
``` javascript
const  responseCodes = {
 httpSuccess: 200,
 httpFailure: 400
} 
```

instead here?

Reply:
Sure, thing.